src/ripple/app/ledger/Ledger.cpp

```diff
@@ -273,6 +287,12 @@ Ledger::Ledger (Ledger const& prevLedger,
     info_.closeTimeResolution = getNextLedgerTimeResolution(
         prevLedger.info_.closeTimeResolution,
         getCloseAgree(prevLedger.info()), info_.seq);
+
+    if (stateMap_->get_version() == SHAMap::version{2})
```

Comment:
alternatively, `stateMap_.is_v2()` ...which `get_version()` seems to be calling anyhow.